clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -131,6 +131,14 @@
             Collections.singletonList(new PartitionInfo(topic, 0, null, null, null)),
             Collections.emptySet(),
             Collections.emptySet());
+
+    private final Cluster oneLeaderCluster = new Cluster(
```

Comment:
Pardon me, why this change is required.

Reply:
producer.close(Duration.of(0))  will force close without sending all message . 
producer.close()  will gracefully close after sending all message .

the callback method will be invoke after  one of above  does

when we want to  send message  , we need to have
-  A Cluster has a leader partition 
-  Mockclient  prepare a fake response

So I add a cluster with one leader

Reply:
if the message is not sent  , the thread will have a infinite loop

```
#Sender 
while(  ... .  this.accumulator.hasUndrained() || this.client.inFlightRequestCount() > 0  ...)
```

Reply:
Thanks for your explanation. ```oneLeaderCluster``` is used by this test case only. Could you make it be a local variable?

Reply:
you are so nice !